// reka-ui/dist/MonthPicker/MonthPickerGrid.js
import { Primitive } from "../Primitive/Primitive.js";
import { injectMonthPickerRootContext } from "./MonthPickerRoot.js";
import { Fragment, computed, createCommentVNode, createElementBlock, createVNode, defineComponent, mergeProps, openBlock, renderSlot, unref, withCtx } from "vue";

//#region src/MonthPicker/MonthPickerGrid.vue?vue&type=script&setup=true&lang.ts
var MonthPickerGrid_vue_vue_type_script_setup_true_lang_default = /* @__PURE__ */ defineComponent({
	__name: "MonthPickerGrid",
	props: {
		asChild: {
			type: Boolean,
			required: false
		},
		as: {
			type: null,
			required: false,
			default: "table"
		}
	},
	setup(__props) {
		const props = __props;
		const rootContext = injectMonthPickerRootContext();
		const disabled = computed(() => rootContext.disabled.value ? true : void 0);
		const readonly = computed(() => rootContext.readonly.value ? true : void 0);
		return (_ctx, _cache) => {
			return openBlock(), createElementBlock(Fragment, null, [createCommentVNode("\n    role=\"application\" is intentional: it ensures screen readers like NVDA pass\n    keyboard events (arrow keys) to the web app instead of intercepting them\n    for virtual cursor navigation. This is the same pattern used by React Aria.\n  "), createVNode(unref(Primitive), mergeProps(props, {
				tabindex: "-1",
				role: "application",
				"aria-labelledby": unref(rootContext).headingId,
				"aria-readonly": readonly.value,
				"aria-disabled": disabled.value,
				"data-readonly": readonly.value && "",
				"data-disabled": disabled.value && ""
			}), {
				default: withCtx(() => [renderSlot(_ctx.$slots, "default")]),
				_: 3
			}, 16, [
				"aria-labelledby",
				"aria-readonly",
				"aria-disabled",
				"data-readonly",
				"data-disabled"
			])], 2112);
		};
	}
});

//#endregion
//#region src/MonthPicker/MonthPickerGrid.vue
var MonthPickerGrid_default = MonthPickerGrid_vue_vue_type_script_setup_true_lang_default;

//#endregion
export { MonthPickerGrid_default };
